webviews/components/history_view.svelte

```diff
@@ -1,16 +1,198 @@
 <script lang="ts">
-  import type { temporal } from "@temporalio/proto"
-  export let history: temporal.api.history.v1.IHistory
+  import { temporal, google } from "@temporalio/proto"
+
+  export let history: temporal.api.history.v1.History
+  type Timestamp = google.protobuf.ITimestamp
+
+  let startTime: number
+
+  interface WorkflowTask {
+    scheduled?: Date
+    started?: Date
+    startedEventId?: number
+    events: temporal.api.history.v1.IHistoryEvent[]
+    status: "COMPLETED" | "FAILED" | "TIMED_OUT"
+    current?: boolean
+  }
+
+  export function tsToDate(ts: Timestamp | null | undefined): Date {
+    if (ts === undefined || ts === null) {
+      throw new Error(`Expected timestamp, got ${ts}`)
+    }
+    const { seconds, nanos } = ts
+    return new Date((seconds as unknown as number) * 1000 + (nanos || 0) / 1000000)
+  }
+
+  //Time conversion to H:M:S
+  //TODO: Have to fix the time to better formatting
+  function timeConvertion(seconds: number): string {
+    if (seconds < 1) {
+      return `${seconds.toFixed(3)}s`
+    }
+    let hour = seconds / 3600
+    let minutes = (hour % 1) * 60
+    let second = (minutes % 1) * 60
+
+    return `${Math.floor(hour)}:${Math.floor(minutes)}:${Math.round(second)}`
+  }
+
+  //Duration calculation
+  function durationCalculation(eventTime: Timestamp | null | undefined, startTime: number): string {
+    return `${timeConvertion(+(tsToDate(eventTime).getTime() / 1000 - startTime))}`
+  }
+
+  //Collecting workflow
+  function workflowTasks(history: temporal.api.history.v1.IHistory): WorkflowTask[] {
+    if (history === undefined) {
+      return []
+    }
+
+    let currWFT: WorkflowTask = { events: [], status: "COMPLETED", current: true }
+    let wfts = Array<WorkflowTask>()
+
+    for (const ev of history.events ?? []) {
```

Comment:
This needs closer review but I'll just handle this myself, requires more intimate knowledge of Temporal.